extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -322,17 +326,32 @@ export class AmpStoryPage extends AMP.BaseElement {
     this.maybeConvertCtaLayerToPageOutlink_();
   }
 
-  /** @private */
-  maybeSetPreviewDuration_() {
-    if (this.storeService_.get(StateProperty.PREVIEW_STATE)) {
-      const videos = this.getAllVideos_();
-
-      const autoAdvanceAttr =
-        videos.length > 0
-          ? VIDEO_PREVIEW_AUTO_ADVANCE_DURATION
-          : DEFAULT_PREVIEW_AUTO_ADVANCE_DURATION;
-
-      this.element.setAttribute('auto-advance-after', autoAdvanceAttr);
+  /**
+   * Configures the page to auto advance using preview-specific durations.
+   * @private
+   */
+  setupAutoAdvanceForPreview_() {
+    const firstVideo = this.getFirstAmpVideo_();
+    const autoAdvanceDuration = firstVideo
+      ? VIDEO_PREVIEW_AUTO_ADVANCE_DURATION
+      : DEFAULT_PREVIEW_AUTO_ADVANCE_DURATION;
+    this.element.setAttribute('auto-advance-after', autoAdvanceDuration);
+
+    if (firstVideo) {
+      whenUpgradedToCustomElement(firstVideo)
+        .then(() => firstVideo.getImpl())
+        .then((videoImpl) => {
+          const loopVideoIfTooShort = (vidLength) =>
+            videoImpl.loop(vidLength < VIDEO_PREVIEW_AUTO_ADVANCE_DURATION_S);
+
+          if (videoImpl.getDuration()) {
+            loopVideoIfTooShort(videoImpl.getDuration());
+          } else {
+            listenOnce(firstVideo, VideoEvents_Enum.LOADEDMETADATA, () => {
+              loopVideoIfTooShort(videoImpl.getDuration());
+            });
```

Comment:
I could simplify this logic and just loop every video in preview mode regardless of length, since longer videos will just progress to the next page after 5 seconds whether or not the video has the 'loop' attribute